src/errordialoghandler.cpp

```diff
@@ -149,7 +160,44 @@ void ErrorDialogHandler::errorDialog(ErrorDialogProperties* pProps) {
     if (!props->m_details.isEmpty()) {
         pMsgBox->setDetailedText(props->m_details);
         if (props->m_detailsUseMonospaceFont) {
-            pMsgBox->setStyleSheet("QTextEdit { font-family: monospace; }");
+            // There is no event to respond on the Show Details button of QMessagBox.
+            // Therefore we must consider the expanded size for positioning the dialog initially.
+            const auto* const pScreen =
+                    mixxx::widgethelper::getScreen(*pMsgBox);
```

Comment:
Unfortunately this is not working satisfyingly for me with Qt 5.12.8 on Ubuntu Studio 20.04 (I'll switch to 22.04 after the first point release is out)
I'm afraid if you pass pMsgBox (which has no parent widget) to `getScreen` this will always return a nullptr with Qt < 5.14.0, thus always display a small popup in the top left corner.
See https://github.com/mixxxdj/mixxx/blob/495225fcbac0cac330895afe814f8dbfbc17fca2/src/util/widgethelper.h#L34L51
I consider this a usability regression compared to the default automatic QMessageBox display.

We could use `qGuiApp->primaryScreen()` in widgethelper.h
That is also used in DlgPrefInterface::getScreen https://github.com/mixxxdj/mixxx/blob/495225fcbac0cac330895afe814f8dbfbc17fca2/src/preferences/dialog/dlgprefinterface.cpp#L211-L221
while in basetracktablemodel.cpp it is considered an "ugly hack" in https://github.com/mixxxdj/mixxx/blob/main/src/library/basetracktablemodel.cpp#L525-L532

It would be interesting how the default placement is implemented in QMessageBox (how it determines the screen) since that was working earlier _without_ a msgBox parent (for single-screen setups).

Reply:
@JoergAtGithub It feels a bit strange if you just restore the current `main` implementation without any comment.
Now the benefit of this PR is available only for qt 5.14+ and lower versions are stuck with current unhandy dialog.

Did you take a look at the places where `qGuiApp->primaryScreen();` is used?
Do you see any reason to _not_ use it here?

Reply:
Done!